apps/src/templates/HintsDisplay.jsx

```diff
@@ -31,6 +33,30 @@ module.exports = React.createClass({
     });
   },
 
+  renderBlocklyHint: function (hint) {
+    var node = document.getElementById(hint.hintId);
+    // Only render if the node exists in the DOM
+    if (node) {
+      Blockly.BlockSpace.createReadOnlyBlockSpace(node, hint.block);
+    }
```

Comment:
An ideal React component should have inputs of state and props, and should output an HTML element. With this design, we now also depend on the current state of the DOM, and then also have side effects outside of our render method. This means I can no longer understand this component in isolation.

I'm not sure I immediately know of a better way of doing this. I'll see if anything comes to mind as I continue perusing the PR :)